lib/elixir/lib/map.ex

```diff
@@ -211,13 +220,16 @@ defmodule Map do
   """
   @spec take(map, [key]) :: map
   def take(map, keys) do
-    Enum.reduce(keys, [], fn key, acc ->
-      case fetch(map, key) do
-        {:ok, value} -> [{key, value} | acc]
-        :error -> acc
-      end
-    end)
-    |> :maps.from_list
+    do_take(keys, map, [])
+  end
```

Comment:
I know it's documented that `take/2` requires list for keys, but before changes any enumerable was accepted. This could be considered a breaking change.